modules/pam_access/pam_access.c

```diff
@@ -599,14 +599,25 @@ netgroup_match (pam_handle_t *pamh, const char *netgroup,
 static int
 user_match (pam_handle_t *pamh, char *tok, struct login_info *item)
 {
-    char   *string = item->user->pw_name;
+    char  *string = NULL;
     struct login_info fake_item;
     char   *at;
     int    rv;
+    char buf[BUFSIZ];
 
     if (item->debug)
       pam_syslog (pamh, LOG_DEBUG,
-		  "user_match: tok=%s, item=%s", tok, string);
+		  "user_match: tok=%s, item=%s", tok, item->user->pw_name);
+
+     memset(buf, 0, BUFSIZ);
+     /* tok is uid */
+     if (strspn(tok, "0123456789") == strlen(tok)) {
+	 sprintf(buf, "%d", item->user->pw_uid);
+	 if (item->debug)
+		pam_syslog(pamh, LOG_DEBUG, "user_match: tok=%s, uid=%d", tok, item->user->pw_uid);
+	 string = buf;
+     } else
+	 string = item->user->pw_name;
```

Comment:
This adds a code that will be executed in all cases including those where it won't be used.
Let's create a new function, e.g. user_or_uid_match, and use it in instead of
```
if ((rv=string_match (pamh, tok, string, item->debug)) != NO)
```

Reply:
The line in question above `if ((rv=string_match (pamh, tok, string, item->debug)) != NO)`
https://github.com/linux-pam/linux-pam/blob/a3c70778c49f1b0ebb93ea2c51df8fc72fe076eb/modules/pam_access/pam_access.c#L659
is not modified as part of this PR.  Can you clarify on this comment?

Reply:
The only place in `user_match()` where uid support is applicable is exactly that `string_match()` invocation I'm talking about, so I suggest changing that invocation instead of adding the code unconditionally in the beginning of `user_match()`.

Reply:
Thank you for the clarification.  Is the uid support not also applicable to the call to `group_match()`?  If a uid is provided, the uid is passed instead of the name.
https://github.com/linux-pam/linux-pam/blob/a3c70778c49f1b0ebb93ea2c51df8fc72fe076eb/modules/pam_access/pam_access.c#L633

Reply:
@ldv-alt thoughts on the above comment?

Reply:
@t8m @ldv-alt thoughts on the above comment?

Reply:
Yes, numeric uid support is also applicable to that call to `group_match`, but this patch doesn't add such support.

Reply:
Added new method user_name_or_uid_match.  I'll resolve this conversation, let me know if there is additional feedback.